File: sample-jdbc-java/src/main/java/com/ucsoftworks/leafdb/jdbc/SampleHelper.java
package com.ucsoftworks.leafdb.jdbc;

import com.ucsoftworks.leafdb.LeafDbSchemaMigration;

import org.jetbrains.annotations.NotNull;

/**
 * Created by Pasenchuk Victor on 29/08/2017
 */

public class SampleHelper extends LeafDbJdbcHelper {

    public SampleHelper(String dbName, int version) {
        super(dbName, version);
    }

    @NotNull
    @Override
    public LeafDbSchemaMigration onCreate() {
        return new LeafDbSchemaMigration().createTable("test");
    }

    @NotNull
    @Override
    public LeafDbSchemaMigration onUpgradeSchema(int oldVersion, int newVersion) {
        return new LeafDbSchemaMigration();
    }
}

File: sample-jdbc/src/main/java/com/ucsoftworks/leafdb/SampleHelper.kt
package com.ucsoftworks.leafdb

import com.ucsoftworks.leafdb.jdbc.LeafDbJdbcHelper

/**
 * Created by Pasenchuk Victor on 28/08/2017
 */
class SampleHelper(dbName: String, version: Int) : LeafDbJdbcHelper(dbName, version) {

    override fun onCreate(): LeafDbSchemaMigration =
            LeafDbSchemaMigration()
                    .createTable(Tables.TEST.name)


    override fun onUpgradeSchema(oldVersion: Int, newVersion: Int): LeafDbSchemaMigration =
            when (oldVersion) {
                1 -> LeafDbSchemaMigration().createTable(Tables.LOAD.tableName)
                else -> LeafDbSchemaMigration()
            }

}
